cmd/gen-go-sample/inittree.go

```diff
@@ -106,6 +109,29 @@ func (t *initTree) get(k string, info pbinfo.Info) (*initTree, error) {
 		} else {
 			// type is a message
 			v.typ.desc = typ
+
+			if d, ok := typ.(*descriptor.DescriptorProto); ok {
+				if d.GetOptions().GetMapEntry() == true {
+					// type is a map entry
+					for _, f2 := range d.Field {
+						if f2.GetName() == "key" {
+							// key is always a primitive type
+							v.typ.keyType = &initType{prim: f2.GetType()}
+						}
+						if f2.GetName() == "value" {
+							if tn2 := f.GetTypeName(); tn2 == "" {
+								// value is a primitive type
+								v.typ.valueType = &initType{prim: f2.GetType()}
+							} else if typ := info.Type[tn2]; typ == nil {
+								return nil, errors.E(nil, "cannot find descriptor of %q", f.GetTypeName())
```

Comment:
Do you mean `f2.GetTypeName()` here instead of `f`?

Reply:
Yes!